aws-janitor/resources/internet_gateways.go

```diff
@@ -77,36 +74,46 @@ func (InternetGateways) MarkAndSweep(opts Options, set *Set) error {
 				break
 			}
 
-			var publicIPsToRelease []*string
+			if opts.DisassociatePublicIP {
+				var publicIPsToRelease []*string
 
-			pageFunc := func(page *ec2.DescribeNetworkInterfacesOutput, _ bool) bool {
-				for _, eni := range page.NetworkInterfaces {
-					publicIPsToRelease = append(publicIPsToRelease, eni.Association.PublicIp)
+				pageFunc := func(page *ec2.DescribeNetworkInterfacesOutput, _ bool) bool {
+					for _, eni := range page.NetworkInterfaces {
+						publicIPsToRelease = append(publicIPsToRelease, eni.Association.PublicIp)
+					}
+					return true
 				}
-				return true
-			}
 
-			if err := svc.DescribeNetworkInterfacesPages(&ec2.DescribeNetworkInterfacesInput{
-				Filters: []*ec2.Filter{
-					{
-						Name:   aws.String("vpc-id"),
-						Values: []*string{aws.String(*att.VpcId)},
+				if err := svc.DescribeNetworkInterfacesPages(&ec2.DescribeNetworkInterfacesInput{
+					Filters: []*ec2.Filter{
+						{
+							Name:   aws.String("vpc-id"),
+							Values: []*string{aws.String(*att.VpcId)},
+						},
 					},
-				},
-			}, pageFunc); err != nil {
-				return err
-			}
+				}, pageFunc); err != nil {
+					logger.Warningf("fail to get public IP for vpc %s: %v", *att.VpcId, err)
+				}
 
-			// According to https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Internet_Gateway.html#detach-igw
-			// Before detaching the internet gateway, we must dissassociate elastic IPs first.
-			for _, publicIP := range publicIPsToRelease {
-				disassociateReq := &ec2.DisassociateAddressInput{
-					PublicIp: aws.String(*publicIP),
+				logger.Warningf("%s: disassociating public IPs for vpc: %s", i.ARN(), *att.VpcId)
+				if opts.DryRun {
```

Comment:
Move dry run check here to better testing the DescribeNetworkInterfacesPages logic in dry run env